app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java

```diff
@@ -48,65 +50,118 @@ private void init() {
         showAvatar();
     }
 
-    public void setupWithNodePubKey(@NonNull String nodePubKey, boolean includeQRCode) {
-        mNodePubKey = nodePubKey;
+    public void setupWithNodeUri(LightningNodeUri nodeUri, boolean includeQRCode) {
+        LightningNodeUri[] tempNodeUris = new LightningNodeUri[1];
+        tempNodeUris[0] = nodeUri;
+
+        setupWithNodeUris(tempNodeUris, includeQRCode);
+    }
+
+    public void setupWithNodeUris(LightningNodeUri[] nodeUris, boolean includeQRCode) {
+        reset();
+        mNodeUris = nodeUris;
+        mIsQRCodeIncluded = includeQRCode;
+
+        /*
+        if (mIsQRCodeIncluded) {
+            mIvQRCode.setOnClickListener(new OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    showAvatar();
+                    ((MotionLayout) findViewById(R.id.userAvatarMotionLayout)).transitionToStart();
+                    if (mListener != null) {
+                        mListener.onHide();
+                    }
+                }
+            });
+
+            mIvUserAvatar.setOnClickListener(new OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    showQRCode();
+                    ((MotionLayout) findViewById(R.id.userAvatarMotionLayout)).transitionToEnd();
+                    if (mListener != null) {
+                        mListener.onReveal();
+                    }
+                }
+            });
+        }
+        */
+
+
         showAvatar();
+        showTorIdentity();
+    }
 
-        if (mNodePubKey != null) {
-            if (!mNodePubKey.isEmpty()) {
-                if (includeQRCode) {
+    public void showTorIdentity() {
+        if (mNodeUris != null) {
+            if (mNodeUris.length > 1) {
+                for (int i = 0; i < mNodeUris.length; i++) {
+                    if (mNodeUris[i].getHost() != null && mNodeUris[i].getHost().toLowerCase().contains("onion")) {
+                        showIdentity(i);
+                        return;
+                    }
+                    showIdentity(0);
+                }
+            } else {
+                showIdentity(0);
+            }
+        }
+    }
+
+    public void showClearNetIdentity() {
```

Comment:
This looks almost exactly like `showTorIdentity`, maybe they can be merged by using an additional parameter?